src/Storages/StorageDistributed.cpp

```diff
@@ -723,57 +722,15 @@ std::optional<QueryProcessingStage::Enum> StorageDistributed::getOptimizedQueryP
     return QueryProcessingStage::Complete;
 }
 
-static bool requiresObjectColumns(const ColumnsDescription & all_columns, ASTPtr query)
-{
-    if (!hasDynamicSubcolumnsDeprecated(all_columns))
-        return false;
-
-    if (!query)
-        return true;
-
-    RequiredSourceColumnsVisitor::Data columns_context;
-    RequiredSourceColumnsVisitor(columns_context).visit(query);
-
-    auto required_columns = columns_context.requiredColumns();
-    for (const auto & required_column : required_columns)
-    {
-        auto name_in_storage = Nested::splitName(required_column).first;
-        auto column_in_storage = all_columns.tryGetPhysical(name_in_storage);
-
-        if (column_in_storage && column_in_storage->type->hasDynamicSubcolumnsDeprecated())
-            return true;
-    }
-
-    return false;
-}
-
 StorageSnapshotPtr StorageDistributed::getStorageSnapshot(const StorageMetadataPtr & metadata_snapshot, ContextPtr query_context) const
 {
     return getStorageSnapshotForQuery(metadata_snapshot, nullptr, query_context);
 }
 
 StorageSnapshotPtr StorageDistributed::getStorageSnapshotForQuery(
```

Comment:
Let's remove this method from `IStorage`. It was the only place, where it differed from `getStorageSnapshot`.